google/cloud/bigtable/tests/instance_admin_emulator.cc

```diff
@@ -136,7 +142,14 @@ class InstanceAdminEmulator final
     if (i == instances_.end()) {
       return grpc::Status(grpc::StatusCode::NOT_FOUND, "instance missing");
     }
-    instances_.erase(i);
+    instances_.erase(i->first);
+
+    for (auto it : clusters_) {
```

Comment:
You probably want `auto const&` there, not that this needs to be super-efficient, but why not?

Reply:
On second thought: this is using the O(N*lgN) algorithm to erase the clusters, you could use:

```C++
// Use something nicer than decltype() here.
std::remove_if(clusters_.begin(), clusters_.end(), [&request](decltype(clusters_)::value_type const& element) {
    return 0 == element.first.find(request->name());
});
```

That is O(N) and clear in its intent.